internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -460,6 +470,147 @@ func (r *BMACReconciler) findBMH(ctx context.Context, agent *adiiov1alpha1.Agent
+		return reconcileComplete{}
+	}
+
+	cdKey := types.NamespacedName{
+		Namespace: agent.Spec.ClusterDeploymentName.Namespace,
+		Name:      agent.Spec.ClusterDeploymentName.Name,
+	}
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	if err := r.Get(ctx, cdKey, clusterDeployment); err != nil {
+		r.Log.WithError(err).Errorf("failed to get clusterDeployment resource %s/%s", cdKey.Namespace, cdKey.Name)
+		return reconcileError{err}
+	}
+
+	// If the cluster is not installed yet, we can't get kubeconfig for the cluster yet.
+	if !clusterDeployment.Spec.Installed {
+		r.Log.Infof("ClusterDeployment %s/%s for agent %s/%s is not installed yet", clusterDeployment.Namespace, clusterDeployment.Name, agent.Namespace, agent.Name)
+		// If cluster is not Installed, wait until a reconcile is trigged by a watch event
+		return reconcileComplete{}
+	}
+
+	// Secret contains kubeconfig for the spoke cluster
+	secret := &corev1.Secret{}
+	name := fmt.Sprintf(adminKubeConfigStringTemplate, clusterDeployment.Name)
+	err := r.Get(ctx, types.NamespacedName{Namespace: clusterDeployment.Namespace, Name: name}, secret)
+	if err != nil && errors.IsNotFound(err) {
+		r.Log.WithError(err).Errorf("failed to get secret resource %s/%s", clusterDeployment.Namespace, name)
+		// If secret is not found, wait until a reconcile is trigged by a watch event
+		return reconcileComplete{}
+	} else if err != nil {
+		return reconcileError{err}
+	}
+
+	spokeClient, err := getSpokeClient(secret)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to create spoke kubeclient")
+		return reconcileError{err}
+	}
+
+	machine, err := r.ensureSpokeMachine(ctx, spokeClient, bmh, clusterDeployment)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to create or update spoke Machine")
+		return reconcileError{err}
+	}
+
+	_, err = r.ensureSpokeBMH(ctx, spokeClient, bmh, clusterDeployment, machine)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to create or update spoke BareMetalHost")
+		return reconcileError{err}
+	}
+
+	return reconcileComplete{}
+}
+
+func (r *BMACReconciler) ensureSpokeBMH(ctx context.Context, spokeClient client.Client, bmh *bmh_v1alpha1.BareMetalHost, clusterDeployment *hivev1.ClusterDeployment, machine *machinev1beta1.Machine) (*bmh_v1alpha1.BareMetalHost, error) {
+	bmhSpoke := &bmh_v1alpha1.BareMetalHost{}
+	err := spokeClient.Get(ctx, types.NamespacedName{Name: bmh.Name, Namespace: bmh.Namespace}, bmhSpoke)
+	if err != nil && errors.IsNotFound(err) {
+		bmhSpoke = &bmh_v1alpha1.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      bmh.Name,
+				Namespace: bmh.Namespace,
+			},
+		}
+		bmhSpoke.Spec = bmh.Spec
+		bmhSpoke.Spec.ExternallyProvisioned = true
+		bmhSpoke.Spec.ConsumerRef.Name = machine.Name
+		bmhSpoke.Spec.ConsumerRef.Namespace = machine.Namespace
+		// copy annotations. hardwaredetails annotations is needed for automatic csr approval
+		bmhSpoke.ObjectMeta.Annotations = bmh.ObjectMeta.Annotations
```

Comment:
i think that HW annotation will not be here.
@flaper87 am i right?

Reply:
The HW details is populated from the agent status inventory. https://github.com/openshift/assisted-service/blob/master/internal/controller/controllers/bmh_agent_controller.go#L324

Hostname field was missing, and I've added that.

Reply:
cool but we miss one more annotation , 
inspect.metal3.io: disabled

Reply:
> cool but we miss one more annotation ,
> inspect.metal3.io: disabled

Do you mean this one? 

https://github.com/openshift/assisted-service/blob/834faecd31fbccaa9716ca020da405a2d13296f8/internal/controller/controllers/bmh_agent_controller.go#L385

Reply:
yep, missed it